File: CodingTestsObjc/TestClass.m
//
//  TestClass.m
//  Macoun-2018objc
//
//  Created by Marcel Weiher on 07.04.20.
//  Copyright © 2020 metaobject. All rights reserved.
//

#import "TestClass.h"

@implementation TestClass

-initWithHi:(long)newHi there:(long)newThere comment:(NSString*)newComment
{
    if ( self=[super init]) {
        _hi = newHi;
        _there = newThere;
        self.comment = newComment;
    }
    return self;
}

-(NSString *)description
{
    return [NSString stringWithFormat:@"<%@:%p: hi: %ld there: %ld \"%@\">",[self class],self,self.hi,self.there,self.comment];
}

-(instancetype)initWithCoder:(NSCoder*)aCoder
{
    self.hi = [aCoder decodeIntForKey:@"hi"];
    self.there = [aCoder decodeIntForKey:@"there"];
    self.comment = [aCoder decodeObjectForKey:@"comment"];
    return self;
}


@end

File: Swacoun-2018/TestClass.swift
//
//  TestClass.swift
//  Swacoun-2018
//
//  Created by Marcel Weiher on 9/14/18.
//  Copyright © 2018 metaobject. All rights reserved.
//

import Foundation
import MPWFoundation

@objc class TestClass: NSObject, Codable {
    let hi:Int32
    let there:Int32
    let comment:String


    init( hi:Int32 , there:Int32, comment:String ) {
        self.hi=hi
        self.there=there
        self.comment=comment
    }
    
    
    public func asDictionary() -> [String:Any] {
        return ["hi": hi,
                "there": there,
                "comment": comment]
    }
    
    @objc override public func write(onJSONStream aStream: MPWJSONWriter!) {
        aStream.writeDictionaryLike( "") { (writer) in
            if let writer=writer {
                writer.writeInteger(self.hi, forKey: "hi")
                writer.writeInteger(self.there, forKey: "there")
                writer.write(self.comment, forKey: "comment")
            }
        }
    }

}
